### PR TITLE
DEV | Property Access Controller

### PR DESCRIPTION
**Pull request by bseptember**
Overview

Added middleware unit test helper.
Removed try catch blocks.
Removed redundant if statements.
Fixed unit tests accordingly.

[Azure board](https://dev.azure.com/retro-rabbit/RetroGradOnboard/_sprints/taskboard/RetroGradOnboard%20Team/RetroGradOnboard/Sprint%2024?workitem=9662)

image

### CODE REVIEW

**Review by MathildaShongwe15 — DISMISSED** [review]

**Review by Juanro-Minne — DISMISSED** [review]

**Review by madwizzard — DISMISSED** [review]

File: RR.App/Controllers/HRIS/PropertyAccessController.cs [hunk 1]
```diff
@@ -24,101 +25,54 @@ public PropertyAccessController(AuthorizeIdentity identity, IPropertyAccessServi
...
-        catch (Exception ex)
-        {
-            return NotFound(ex.Message);
+            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey")
+            {
+                var employee = await _employeeService.GetEmployee(email);
+                return Ok(employee!.Id);
+            }
         }
+        throw new CustomException("Error retrieving employee.");
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/PropertyAccessController.cs:76]: Reducing the nested if statements can make it easier to read. I'm suggesting an additional check to see if employee exist before we return their id. Prevents null reference exceptions.

```
if (string.IsNullOrEmpty(_identity.Email)) 
    throw new CustomException("Error retrieving employee.");

if (email == _identity.Email)
    return Ok(_identity.EmployeeId);

if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey"))
    throw new CustomException("Error retrieving employee.");

var employee = await _employeeService.GetEmployee(email);
return employee == null 
    ? throw new CustomException("Error retrieving employee.") 
    : Ok(employee.Id);
```

File: RR.App.Tests/Controllers/HRIS/PropertyAccessControllerUnitTests.cs [hunk 2]
```diff
@@ -5,22 +5,27 @@
...
 
 namespace RR.App.Tests.Controllers.HRIS;
 
 public class PropertyAccessControllerUnitTests
 {
-
     private readonly Mock<IPropertyAccessService> _propertyAccessMockService;
     private readonly Mock<IEmployeeService> _employeeMockService;
     private readonly PropertyAccessController _propertyAccessController;
+    private readonly Mock<AuthorizeIdentity> _authorizeIdentityMock;
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/PropertyAccessControllerUnitTests.cs:21]: This variable is referenced but never used.

File: RR.App.Tests/Controllers/HRIS/PropertyAccessControllerUnitTests.cs [hunk 3]
```diff
@@ -5,22 +5,27 @@
...
 
-    public PropertyAccessControllerUnitTests() {
+    public PropertyAccessControllerUnitTests()
+    {
+        _authorizeIdentityMock = new Mock<AuthorizeIdentity>(MockBehavior.Strict, null, null);
         _propertyAccessMockService = new Mock<IPropertyAccessService>();
         _employeeMockService = new Mock<IEmployeeService>();
-        _propertyAccessController = new PropertyAccessController(new AuthorizeIdentityMock(), _propertyAccessMockService.Object, _employeeMockService.Object);
+        _propertyAccessController = new PropertyAccessController(new AuthorizeIdentityMock(1), _propertyAccessMockService.Object, _employeeMockService.Object);
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/PropertyAccessControllerUnitTests.cs:28]: Either use `AuthorizeIdentityMock` or `Mock<AuthorizeIdentity>`  in your tests. I see you using both. 

Just for consistency.

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by madwizzard — DISMISSED** [review]

**Review by Juanro-Minne — DISMISSED** [review]

**Review by EduanO — DISMISSED** [review]

**Review by MathildaShongwe15 — DISMISSED** [review]

**Review by KAYEMBAJONATHAN — DISMISSED** [review]

**Review by KAYEMBAJONATHAN — APPROVED** [review]

**Review by MathildaShongwe15 — APPROVED** [review]

**Comment by bseptember** [comment]
Thank you @MathildaShongwe15  for pointing out the other controller instantiation which was unnecessary.👍

**Review by EduanO — APPROVED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Pull request merged**